cmd/agent/main.go

```diff
@@ -5,7 +5,8 @@
 
 // +build !windows,!android
 
-//go:generate go run ../../pkg/config/render_config.go agent ../../pkg/config/config_template.yaml ./dist/datadog.yaml
+//go:generate go run ../../pkg/config/render_config.go agent-py2py3 ../../pkg/config/config_template.yaml ./dist/datadog.yaml
```

Comment:
I'm missing the point here: why do we generate the configuration twice ? Won't the second call overwrite the file ?

Reply:
See my [previous comment](https://github.com/DataDog/datadog-agent/pull/4013/files/b160c624b5a92ce06bc02b90f5dcae19d03322b2#r313784940): the goal is to selectively generate one command or the other depending on which python versions we are including.
The solution I have right now is far from perfect (it generates the file three times), but keeps the current behaviour (all `go:generate` directives put in the `cmd/agent` folder are automatically run without needing to change something). Another solution would be to explicitly list all directives that would be run, but that means each time someone wants to add / remove a directive, they must add it to the list of generated directives.

Either solution is fine to me, but WDYT?

Reply:
how many other `go:generate` directives are run when invoke runs `go generate -x <REPO_PATH>/cmd/agent`?

Depending on how many are run, we could either:

* if there aren't many, stop using `go generate` and simply make the invoke task execute the `go run render_config.go` as many times as needed, with the right arguments
* if there are many, only remove the generation of `datadog.yaml` from `go:generate`, and run this generation only with invoke directly (`go run render_config.go [...]`

Overall I don't feel the feature for which we use `go:generate` justify the extra complexity you're adding here.